lib/src/execution/windows/execution.cc

```diff
@@ -195,11 +195,151 @@ namespace facter { namespace execution {
         return commandline;
     }
 
-    pair<bool, string> execute(
+    // Typedef for pipe data
+    typedef tuple<HANDLE, OVERLAPPED*, bool, string*, function<bool(string const&)> const*> pipe_data;
```

Comment:
I've pushed a commit with it changed to a struct for POSIX (I think it's cleaner).  Going to make the same changes to Windows now.